docs/architecture/mbed-crypto-storage-specification.md

```diff
@@ -1,9 +1,11 @@
-Mbed Crypto storage specification
+Mbed TLS storage specification
 =================================
 
-This document specifies how Mbed Crypto uses storage.
+This document specifies how Mbed TLS uses storage.
+Key storage was originally introduced in a product called Mbed Crypto, which was re-distributed via Mbed TLS and has since been merged into Mbed TLS.
+This document contains historical information both from before and after this merge.
 
-Mbed Crypto may be upgraded on an existing device with the storage preserved. Therefore:
```

Comment:
Actually, I think this could be done reasonably:

1. (new) line 8 should say `Mbed Crypto/TLS may be upgraded on an existing device with the storage preserved.`
2. leave other references to `Mbed Crypto` except
3. the two `Mbed Crypto TBD` instances (2019 and 2020) should be investigated and details put in when we have them 

only #1 needs be done in this PR (#3 is more work than a rename warrants)